shared/js/l10n.js

```diff
@@ -1371,17 +1387,59 @@
     this._emitter.emit('ready');
   }
 
-  Context.prototype.registerLocales = function(defLocale, available) {
-    /* jshint boss:true */
-    this.availableLocales = [this.defaultLocale = defLocale];
+  function getMatchingLangpack(lpVersions) {
+    for (var i in lpVersions) {
+      if (lpVersions[i].target === navigator.mozL10n.gaiaVersion) {
+        return lpVersions[i];
+      }
+    }
+    return null;
+  }
 
-    if (available) {
-      for (var i = 0, loc; loc = available[i]; i++) {
-        if (this.availableLocales.indexOf(loc) === -1) {
-          this.availableLocales.push(loc);
+  function extendLanguagesList(availableLangs, extraLangs) {
+    var loc, lp;
+    var langs = {};
+
+    if (availableLangs) {
+      for (loc in availableLangs) {
```

Comment:
OK, I now see `splitAvailableLanguagesString`;  `availableLangs` is an object with lang codes and versions.  This changes the Context API.  If we do this, I think we should release a new minor version: 2.1.x or even 2.2.x (to match the Gaia version). I wonder if maybe instead we should either:

 - make `availableLangs` polymorphic: an array for legacy code and an object with version for new code, or
 - accept it only as an array and map `splitAvailableLanguagesString` over its elements;  this would make it inconsistent with the type signature of `extraLangs`, so I'm not sure this is a good idea at all.